### PR TITLE
Enable pipelines by default

### PR DESCRIPTION
**Pull request by jalvz**
related: https://github.com/elastic/apm-server/issues/1283

- [x] needs changelog
- [ ] needs docs update
- [x] consider deprecating `setup pipelines` cmd

### CODE REVIEW

File: beater/beater.go [hunk 1]
```diff
@@ -108,9 +108,7 @@ func New(b *beat.Beat, ucfg *common.Config) (beat.Beater, error) {
 		logger:  logger,
 	}
 
-	// setup pipelines if explicitly directed to or setup --pipelines and config is not set at all
-	shouldSetupPipelines := beaterConfig.Register.Ingest.Pipeline.isEnabled() ||
-		(b.InSetupCmd && beaterConfig.Register.Ingest.Pipeline.Enabled == nil)
+	shouldSetupPipelines := beaterConfig.Register.Ingest.Pipeline.isEnabled() || b.InSetupCmd
```
Comment (simitt) [inline beater/beater.go]: With this change pipeline would always be setup when in `setup` cmd, ignoring the actual setting of `register.ingest.pipeline.enabled`.

**Comment by simitt** [comment]
> consider deprecating setup pipelines cmd

Why would you want to deprecate this? Afaik the setup cmd followed by a specific option should _only_ set this up.

File: beater/beater.go (hunk 1 above)
Comment (jalvz) [reply beater/beater.go]: yes, is it wrong?

**Comment by jalvz** [comment]
> Why would you want to deprecate this

Because if we default to true when starting apm-server, it becomes redundant I guess. And it is simpler to just provide one way of doing things.

**Comment by graphaelli** [comment]
Logstash/Kafka output users would still need a setup command, I do not think we should deprecate.

File: beater/beater.go (hunk 1 above)
Comment (simitt) [reply beater/beater.go]: From my understanding yes. The `setup` command should respect the configuration file, and when `register.ingest.pipeline.enabled` is set to `false`, then why should they be set up?

**Comment by simitt** [comment]
This is my understanding from the `setup` cmd:
APM Server currently has 4 setup options:
- `apm-server setup`: should run all setup cmds that are available, according to the config
- `apm-server setup --template`: only concerns what needs to be set up for templates
- `apm-server setup --ilm-policy`: only concerns what needs to be set up for using ilm and policies
- `apm-server setup --pipelines`: only concerns what needs to be set up for ingest pipelines

File: beater/beater.go (hunk 1 above)
Comment (jalvz) [reply beater/beater.go]: i believe the fix is like e9c50c240b913977e244fd8eefebc5bf40430f31 then

**Comment by simitt** [comment]
This PR fixes https://github.com/elastic/apm-server/issues/1972 (Only the geo-ip pipeline task has been linked so far).

**Comment by jalvz** [comment]
> Since we are trying to move away from having default settings only set in the yml file, I suggest to check if it is possible to enable in code

Good point, doesn't look that it is possible because lists are not merged when unpacking config. So options are:
1 - enable only in config file
2 - document explicitly that if user wants to add another pipeline, they need to enable "apm_user_agent" as well
3 - park this until we have better support for this

**Comment by simitt** [comment]
After taking some time to look into the pipeline configuration, what do you think about going into the same direction as we did with the `idxmgmt.Supporter`? I would not directly couple it to the `idxmgmt` handling as it only concerns output pipelines for ES. Conceptually we could aim for introducing a pipeline supporter with a `BuildSelector` interface in libbeat though, see [current build selector handling in libbeat](https://github.com/elastic/beats/blob/master/libbeat/outputs/elasticsearch/elasticsearch.go#L219). This would allow for implementing our own pipeline handling, similar to the [`idxmgmt.BuildSelector` implementation](https://github.com/elastic/apm-server/blob/master/idxmgmt/supporter.go#L111). If you're up for it we should talk to the beats core team.

**Comment by jalvz** [comment]
Closing this for now, if we decide to not use the supporter approach, I'll reopen again

**Pull request closed without merging**